lib/mosaic-utils/src/main/java/org/eclipse/mosaic/lib/util/gson/UnitFieldAdapter.java

```diff
@@ -167,5 +173,91 @@ private double determineUnitMultiplier(String prefix, String unit) {
             return (TypeAdapter<T>) new UnitFieldAdapter(false, SPEED_PATTERN);
         }
     }
+
+    public static class WeightKiloGrams implements TypeAdapterFactory {
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
+            return (TypeAdapter<T>) new UnitFieldAdapter(true, WEIGHT_PATTERN) {
+                @Override
+                double determineUnitMultiplier(String prefix, String unit) {
+                    if (unit.startsWith("mile")) {
```

Comment:
I guess we do not need that here for weights 😄

Reply:
Yes I forgot to copy/paste the `WeightKiloGramsQuiet` solution 😅